File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 9/16/2025, 12:53:48 AM
class Solution:
    def maxArea(self, height: List[int]) -> int:
        vol = 0
        i = 0 
        j = len(height)-1
        while(i<j):
            cv = (j-i)*min(height[i],height[j])
            vol = max(vol,cv)
            if height[i] > height[j]:
                j -=1
            elif height[j] > height[i]:
                i +=1
            else:
                i +=1
                j -=1
        return vol
        

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 2/13/2026, 12:58:43 PM
class Solution {
public:
    int maxArea(vector<int>& height) 
    {
        int s = 0, e = height.size()-1;
        int v = 0;
        while(s<e)
        {
            int vol = min(height[s],height[e]) * (e-s);
            if(vol>v)
                v = vol;
            if(height[s]>=height[e])
                e--;
            else
                s++;
        }
        return v;
    }
};